development/Dockerfile

```diff
@@ -1,44 +1,71 @@
+# Version: 0.0.1
+#
+# Plugins that need to add additional steps or packages can do in the section below.
+# -------------------------------------------------------------------------------------
+# !!! CAUTION MODIFYING LINES BELOW
+
+# Accepts a desired Nautobot version as build argument, default to 1.2.0
 ARG NAUTOBOT_VER="1.4.0"
-ARG PYTHON_VER=3.7
 
-FROM ghcr.io/nautobot/nautobot-dev:${NAUTOBOT_VER}-py${PYTHON_VER}
+# Accepts a desired Python version as build argument, default to 3.7
+ARG PYTHON_VER="3.7"
 
-# Make the value available after the FROM directive
-ARG NAUTOBOT_VER
-ENV prometheus_multiproc_dir=/prom_cache
+# Retreive published development image of Nautobot base which should include most CI dependencies
+FROM ghcr.io/nautobot/nautobot-dev:${NAUTOBOT_VER}-py${PYTHON_VER}
 
+# Runtime argument and environment setup
 ARG NAUTOBOT_ROOT=/opt/nautobot
 
+ENV prometheus_multiproc_dir=/prom_cache
 ENV NAUTOBOT_ROOT ${NAUTOBOT_ROOT}
 
-WORKDIR $NAUTOBOT_ROOT
-
-# Configure poetry
+# Don't need virtual environments in container, parallelized installer has inconsistent behavior
 RUN poetry config virtualenvs.create false \
     && poetry config installer.parallel false
 
+# !!! CAUTION MODIFYING LINES ABOVE
 # -------------------------------------------------------------------------------------
-# Install Nautobot Plugin
+# Plugin-specifc system build/test dependencies.
+# 
+# Example: LDAP requires `libldap2-dev` to be apt-installed before the Python package.
 # -------------------------------------------------------------------------------------
-WORKDIR /tmp/install
+# --> Start safe to modify section
 
-# Copy in only pyproject.toml/poetry.lock to help with caching this layer if no updates to dependencies
-COPY poetry.lock pyproject.toml /tmp/install/
+# Uncomment the line below if you are apt-installing any package.
+# RUN apt update
+# RUN apt install libldap2-dev
 
-# Add the requested Nautobot version to pyproject
-# to install the correct version based on the NAUTOBOT_VER argument
-# Otherwise Poetry will override the version in this container
-# with the one in the poetry.lock
-RUN poetry add nautobot=${NAUTOBOT_VER}
-
-# --no-root declares not to install the project package since we're wanting to take advantage of caching dependency installation
-# and the project is copied in and installed after this step
-RUN poetry install --no-interaction --no-ansi --no-root
+# --> Stop safe to modify section
+# -------------------------------------------------------------------------------------
+# Install Nautobot Plugin
+# -------------------------------------------------------------------------------------
+# !!! CAUTION MODIFYING LINES BELOW
 
-# Copy in the rest of the source code and install local Nautobot plugin
+# Copy in the source code
 WORKDIR /source
 COPY . /source
-RUN cp /tmp/install/* /source/
-RUN poetry install --no-interaction --no-ansi
+
+# Get container's installed Nautobot version as a forced constraint
```

Comment:
Maybe add a comment explaining that we can't just use `NAUTOBOT_VER` because it might be `latest` rather than a specific version?